libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -241,6 +267,167 @@ void AP_Camera_Backend::send_camera_settings(mavlink_channel_t chan) const
+        }
+
+        float dist_increment_m;
+        if (!terrain->get_terrain_spacing(dist_increment_m)) {
+            // failed to get the terrain spacing
+            continue;
+        }
+
+        float dist_max = 10000; // Hardcoded for testing
+        float total_dist = 0;
+        float mount_pitch_deg = att[1];
+        float mount_yaw_ef_deg = wrap_180(att[2] + degrees(ahrs.get_yaw()));
+        bool get_terrain_alt_success = true;
+        float prev_terrain_amsl_m = terrain_amsl_m;
+        // iteratively move test_loc forward until its alt-above-sea-level is below terrain-alt-above-sea-level
+        while (total_dist < dist_max && (test_loc.alt * 0.01) > terrain_amsl_m) {
+            total_dist += dist_increment_m;
+
+            // Take backup of previous test location and terrain amsl
+            prev_test_loc = test_loc;
+            prev_terrain_amsl_m = terrain_amsl_m;
+
+            // Move test location forward
+            test_loc.offset_bearing_and_pitch(mount_yaw_ef_deg, mount_pitch_deg, dist_increment_m);
+
+            // Get terrain's alt-above-sea-level (at test_loc)
+            // Fail if terrain alt cannot be retrieved
+            if (!terrain->height_amsl(test_loc, terrain_amsl_m, true) || std::isnan(terrain_amsl_m)) {
+                // failed to get terrain alt
+                get_terrain_alt_success = false;
+                continue;
+            }
+        }
+        // if a fail occurs in the loop while getting terrain alt then continue to next interation
+        if (get_terrain_alt_success == false) {
+            continue;
+        }
+        if (total_dist >= dist_max) {
+            // POI: unable to find terrain within dist_max
+            continue;
+        } else if (terrain_amsl_m<=0.0f) {
+            // POI: failed to retrieve terrain alt
+            continue;
+        } else {
+            // test location has dropped below terrain
+            // interpolate along line between prev_test_loc and test_loc
+            float dist_interp_m = linear_interpolate(0, dist_increment_m, 0, prev_test_loc.alt * 0.01 - prev_terrain_amsl_m, test_loc.alt * 0.01 - terrain_amsl_m);
+            {
+                WITH_SEMAPHORE(poi_calculation.sem);
+                poi_calculation.poi_loc = prev_test_loc;
+                poi_calculation.poi_loc.offset_bearing_and_pitch(mount_yaw_ef_deg, mount_pitch_deg, dist_interp_m);
+                poi_calculation.att_quat = {quat[0], quat[1], quat[2], quat[3]};
+                poi_calculation.loc = curr_loc;
+                poi_calculation.poi_update_ms = AP_HAL::millis();
+            }
+        }
+    }
+}
+
+// send camera field of view status
+void AP_Camera_Backend::send_camera_fov_status(mavlink_channel_t chan)
+{
+    // Register a request for calculating poi
+    {
+    WITH_SEMAPHORE(poi_calculation.sem);
+    poi_calculation.req_ms = AP_HAL::millis();
+    poi_calculation.requested = true;
+    poi_calculation.chan = chan;
+    }
+}
```

Comment:
Need to actually send the message in here.  We call into the mount library to get the pointing-at-latn-ng-alt.  It can jot down that it's been requested rather than setting the state here.